serf/merge_delegate.go

```diff
@@ -45,5 +56,28 @@ func (m *mergeDelegate) nodeToMember(n *memberlist.Node) *Member {
 		DelegateMin: n.DMin,
 		DelegateMax: n.DMax,
 		DelegateCur: n.DCur,
+	}, nil
+}
+
+// validateMemberInfo checks that the data we are sending is valid
+func (m *mergeDelegate) validiateMemberInfo(n *memberlist.Node) error {
+	var InvalidNameRe = regexp.MustCompile(`[^A-Za-z0-9\\-]+`)
+
+	if len(n.Name) > 128 {
+		return fmt.Errorf("NodeName length is %v characters. Valid length is between "+
+			"1 and 128 characters.", len(n.Name))
+	}
+	if InvalidNameRe.MatchString(n.Name) {
+		return fmt.Errorf("Nodename contains invalid characters %v , Valid characters include "+
+			"all alpha-numerics and dashes", n.Name)
+	}
+
+	if net.ParseIP(string(n.Addr)) == nil {
+		return fmt.Errorf("Address is %v . Must be a valid representation of an IP address. ", n.Addr)
```

Comment:
We should check for `address:port` format; the field is called Addr but stores the port too.